src/orb.yaml

```diff
@@ -0,0 +1,156 @@
+---
+version: 2.1
+
+definitions:
+  port-parameter: &port-parameter
+    description: port
+    type: integer
+  localhost-parameter: &localhost-parameter
+    description: host
+    type: string
+    default: localhost
+  wait-for-attributes: &wait-for-attributes
+    seconds_between_retries: <<parameters.seconds_between_retries>>
+    timeout: <<parameters.timeout>>
+    description: <<parameters.description>>
+
+description: Waits for things
+
+examples:
+  waiting_for_containers:
+    description: "Waiting for containers to be ready for start testing"
+    usage:
+      version: 2.1
+      orbs:
+        wait-for: cobli/wait-for@x.x.x
+      jobs:
+        build:
+          docker:
+            - image: circleci/openjdk:8
+            - image: cassandra:3.2
+          steps:
+            - checkout
+            - wait-for/cassandra
+            - run:
+                name: Test
+                command: sbt test
+
+commands:
+  sh-command:
+    description: Waits for a shell command to be sucessfully executed
+    parameters: &wait-for-command-parameters
+      description:
+        description: command name
+        type: string
+        default: command
+      timeout:
+        description: timeout
+        type: integer
+        default: 60
+      seconds_between_retries:
+        description: seconds
+        type: integer
+        default: 1
+      sh_command:
+        description: command to be checked
+        type: string
+        default: "true"
+    steps:
+      - run:
+          name: Wait for <<parameters.description>>
+          command: |
+            timeout <<parameters.timeout>> sh -c \
+              'until <<parameters.sh_command>>; do
+                echo "Waiting for <<parameters.description>> ..."
+                sleep <<parameters.seconds_between_retries>>
+              done'
+
+  port:
+    description: Waits for a port being up
+    parameters: &wait-for-port-parameters
```

Comment:
This is not a request but a suggestion. I usually like to declare achors in a more global key, variables maybe.
So it is easier to find the declaration whitout using find command. Just saying

Reply:
I have refactored the file and all definitions are inside `definitions` key.

It's definitively better (I toke `definitions` name from OpenAPI's specs section that have the same objective. I think that is more precise that `variables`)